pkg/reconciler/taskrun/taskrun_test.go

```diff
@@ -3308,34 +3312,63 @@ spec:
         name: pvc2
 `)
 
-	d := test.Data{
-		Tasks:        []*v1.Task{taskWithTwoWorkspaces},
-		TaskRuns:     []*v1.TaskRun{taskRun},
-		ClusterTasks: nil,
-	}
-	testAssets, cancel := getTaskRunController(t, d)
-	defer cancel()
-	clients := testAssets.Clients
-	createServiceAccount(t, testAssets, "default", "foo")
-	_ = testAssets.Controller.Reconciler.Reconcile(testAssets.Ctx, getRunName(taskRun))
+	tcs := []struct {
+		name                string
+		cfgMap              map[string]string
+		expectFailureReason string
+	}{{
+		name: "multiple PVC based Workspaces in per workspace coschedule mode - failure",
+		cfgMap: map[string]string{
+			"disable-affinity-assistant": "false",
+			"coschedule":                 "workspaces",
+		},
+		expectFailureReason: podconvert.ReasonFailedValidation,
+	}, {
+		name: "multiple PVC based Workspaces in per pipelinerun coschedule mode - success",
+		cfgMap: map[string]string{
+			"disable-affinity-assistant": "true",
+			"coschedule":                 "pipelineruns",
+		},
+	}}
 
-	_, err := clients.Pipeline.TektonV1().Tasks(taskRun.Namespace).Get(testAssets.Ctx, taskWithTwoWorkspaces.Name, metav1.GetOptions{})
-	if err != nil {
-		t.Fatalf("krux: %v", err)
-	}
+	for _, tc := range tcs {
+		d := test.Data{
+			Tasks:        []*v1.Task{taskWithTwoWorkspaces},
+			TaskRuns:     []*v1.TaskRun{taskRun},
+			ClusterTasks: nil,
+			ConfigMaps: []*corev1.ConfigMap{{
+				ObjectMeta: metav1.ObjectMeta{Namespace: system.Namespace(), Name: config.GetFeatureFlagsConfigName()},
+				Data:       tc.cfgMap,
+			}},
+		}
+		testAssets, cancel := getTaskRunController(t, d)
+		defer cancel()
+		clients := testAssets.Clients
+		createServiceAccount(t, testAssets, "default", "foo")
+		_ = testAssets.Controller.Reconciler.Reconcile(testAssets.Ctx, getRunName(taskRun))
 
-	ttt, err := clients.Pipeline.TektonV1().TaskRuns(taskRun.Namespace).Get(testAssets.Ctx, taskRun.Name, metav1.GetOptions{})
-	if err != nil {
-		t.Fatalf("expected TaskRun %s to exist but instead got error when getting it: %v", taskRun.Name, err)
-	}
+		_, err := clients.Pipeline.TektonV1().Tasks(taskRun.Namespace).Get(testAssets.Ctx, taskWithTwoWorkspaces.Name, metav1.GetOptions{})
+		if err != nil {
+			t.Fatalf("krux: %v", err)
```

Comment:
nit: can this be replaced with a more informative message? (or even just t.Fatal(err))

Reply:
error message updated